src/themis/secure_session.c

```diff
@@ -826,7 +827,7 @@ static themis_status_t secure_session_finish_client(secure_session_t *session_ct
 		return THEMIS_INVALID_PARAMETER;
 	}
 
-	if (memcmp(proto_message->tag, THEMIS_SESSION_PROTO_TAG, SOTER_CONTAINER_TAG_LENGTH))
+	if (memcmp(proto_message->tag, THEMIS_SESSION_PROTO_TAG, SOTER_CONTAINER_TAG_LENGTH) != 0)
```

Comment:
we do compare public data here, right?

Reply:
actually yes, it's trying to check that it's correct secure session packet or no